openstudiocore/src/model/ThermalZone.cpp

```diff
@@ -1998,18 +1998,13 @@ namespace detail {
   {
     boost::optional<AirLoopHVACSupplyPlenum> result;
 
-    std::vector<AirLoopHVACSupplyPlenum> plenums = model().getModelObjects<AirLoopHVACSupplyPlenum>();
+    std::vector<AirLoopHVACSupplyPlenum> plenums = getObject<ModelObject>().getModelObjectSources<AirLoopHVACSupplyPlenum>();
 
-    for(const auto & plenum : plenums)
-    {
-      if( boost::optional<ThermalZone> tz = plenum.thermalZone() )
-      {
-        if( tz->handle() == handle() )
-        {
-          result = plenum;
-          break;
-        }
-      }
+    if (plenums.size() == 1){
```

Comment:
@macumber If you switch to using `getConcreteModelObjects` you'll need revert to using the old approach.